### PR TITLE
GAWB-2175: reset Submission Details page to 1 on filter button click

### PR DESCRIPTION
**Pull request by jmthibault79**
Also update Submission Details to the New Table

Thanks to Matt P for much help!

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs [hunk 1]
```diff
@@ -27,78 +28,93 @@
...
+       {:items
+        (constantly
+         [[comps/FilterGroupBar
+           {:data (:workflows props)
+            :selected-index (:filter-group-index @state)
+            :on-change (fn [index data]
+                         (swap! state assoc
+                                :filter-group-index index
+                                :filtered-data data)
+                         ((@refs "table") :update-query-params {:page-number 1}))
```
Comment (helgridly) [inline src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs:113]: this looks like it!

**Comment by helgridly** [comment]
Seems legit. I defer to a UI grownup.

**Review by helgridly — APPROVED** [review]

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs [hunk 2]
```diff
@@ -8,7 +8,8 @@
    [broadfcui.common.icons :as icons]
    [broadfcui.common.modal :as modal]
    [broadfcui.common.style :as style]
-   [broadfcui.common.table :refer [Table]]
+   [broadfcui.common.table.table :refer [Table]]
+   [broadfcui.common.table.style :as table-style]
```
Comment (ansingh7115) [inline src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: sorry this going to be super duper nitpicky, but these should all be in alphabetical order. Would you mind rearranging?

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs [hunk 3]
```diff
@@ -27,78 +28,93 @@
...
+                              [:div {}
+                               (map (fn [message]
+                                      [:div {} message])
+                                    message-list)])}
+                   {:header "Workflow ID" :initial-width 300
+                    :as-text :workflowId :sort-by :text
+                    :render
+                    (fn [workflow]
+                      (let [{:keys [submission-id bucketName]} props
+                            inputs (second (second (first (:inputResolutions workflow))))
```
Comment (ansingh7115) [inline src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: Could potentially use a threading macro here like 

`(-> workflow :inputResolutions first second second)`

But I guess the way it is now is pretty clear, so it's not necessary

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs [hunk 4]
```diff
@@ -27,78 +28,93 @@
...
-                 (:messages row)
-                 row])}])
+      {:ref "table"
+       :data (:filtered-data @state)
+       :body
+       {:empty-message "No Workflows"
+        :style table-style/table-heavy
+        :behavior {:fixed-column-count 1}
+        :columns [
+                   {:id "view"
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]:
```clojure
:columns [{:id "view"
```
And reindent from there. In general, an open bracket should never be the last thing on a line, and a close bracket should never be the first thing on a line.

Comment (jmthibault79) [reply src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: ok updated

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs (hunk 2 above)
Comment (jmthibault79) [reply src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: ok updated

**Comment by ansingh7115** [comment]
Looks good to me 👍

**Pull request merged**